src/main/java/christmas/domain/benefit/Gift.java

```diff
@@ -0,0 +1,48 @@
+package christmas.domain.benefit;
+
+import java.util.EnumSet;
+import java.util.Optional;
+
+public enum Gift {
+    CHAMPAGNE("샴페인", 1, 25000, 120000);
+
+    private final String product;
+    private final int quantity;
+    private final int price;
+    private final int minOrderPrice;
+
+    Gift(String product, int quantity, int price, int minOrderPrice) {
+        this.product = product;
+        this.quantity = quantity;
+        this.price = price;
+        this.minOrderPrice = minOrderPrice;
+    }
+
+    public static boolean givable(int totalOrderPrice) {
+        return findMatchingGift(totalOrderPrice).isPresent();
+    }
+
+    public static Gift findByTotalOrderPrice(int totalOrderPrice) {
+        return findMatchingGift(totalOrderPrice).get();
+    }
+
+    public static int findPriceByTotalOrderPrice(int totalOrderPrice) {
+        return findMatchingGift(totalOrderPrice)
+                .map(gift -> gift.price)
+                .orElse(0);
+    }
+
+    private static Optional<Gift> findMatchingGift(int totalOrderPrice) {
+        return EnumSet.allOf(Gift.class).stream()
+                .filter(gift -> totalOrderPrice >= gift.minOrderPrice)
+                .findFirst();
```

Comment:
```suggestion
        return Arrays.stream(values())
                .filter(gift -> totalOrderPrice >= gift.minOrderPrice)
                .findFirst();
```

values()로 enum의 모든 상수들을 배열로 가져오는 방법도 있는 것으로 알고 있어서요!
해당 방법을 쓰신 이유가 있을까요?

Reply:
EnumSet은 Set 구현체로 메모리를 효율적으로 사용하고 연산이 빠르다고 하더라구요
이렇게 데이터가 작을 때는 성능면에서 큰 차이가 없겠지만
Arrays.stream(values()) 를 사용하면 Enum의 모든 값을 배열로 반환하고 이 모든 배열을 모두 스트림으로 변환한다는 점에서
보다 성능이 좋고 특정 클래스를 지원하는 구현체를 사용함으로써 장점을 더욱 살리고 싶어 사용하였습니다!

Reply:
Enum 인스턴스가 과연 많아질지, 많아지는 것이 좋은 설계인지 고민해보면 좋을 것 같습니다!